### ISSUE TITLE
Feature request: Force IP Update

### ISSUE THREAD

**Issue by killerbigpoint** [qdm12/ddns-updater#780]
1. What's the feature?
Been using Noip on the free plan for a while now. Now the problem is that after 30 days of "inactivity" without any IP record updates it will auto delete your dns. Have an option to force update every like 7 days or so, so it can keep operating.

2. Extra information?
Since I use truenas for this, maybe an enviroment option would be nice too but that's up to you

**Comment by Mixa84** [comment]
:+1: for this one. I'm also using NoIP free plan and if there is no ip change then there is no update so need to confirm my host via email link from NoIP periodically.

**Comment by bentemple** [comment]
>👍 for this one. I'm also using NoIP free plan and if there is no ip change then there is no update so need to confirm my host via email link from NoIP periodically.

Can either of you verify that a manual update with the same IP does in-fact reset that clock? I ask because they're (imo) trying to dis-incentivize using the free plan, so it's probably worth confirming that an update even if it's the same IP, does in-fact reset that clock. I could also see it having no impact because the IP is the same. If it does reset that clock, then this makes a lot of sense to me.

**Comment by Mixa84** [comment]
I'm looking at my emails to confirm the hostname and the period of them is not symmetric so not sure how will I check if it works. But also on the other side, how do I do manual update from DDNS Updater (I'm using it on TrueNAS) ?

**Comment by qdm12** [comment]
> Can either of you verify that a manual update with the same IP does in-fact reset that clock?

Yep that would be great. I'm ready to jump in implement this but that's going to be a few hours of programming, it would be nice if someone can confirm sending an update to the existing ip does the trick. Otherwise we could update to the localhost ip address for a hot second amd update back to the actual public ip address, but I would prefer to avoid that.

**Comment by bentemple** [comment]
> But also on the other side, how do I do manual update from DDNS Updater (I'm using it on TrueNAS) ?
I don't believe you'll be able to with ddns-updater, since even if you delete your config, it is going to check the DNS, realize it's set correctly and no-op (I think). You'd need to manually hit their API. A quick curl command should suffice ([api docs](https://www.noip.com/integrate/request)):
```
curl https://$NOIPUSER:$NOIPPASS@dynupdate.no-ip.com/nic/update?hostname=$NOIPHOST&myip=$MYIP"
```
Where NOIPUSER, NOIPPASS, NOIPHOST, and MYIP are set to relevant values.

Frankly, looking at the NO-IP website, I don't think setting the DNS IP again is going to reset the clock:

![screenshot_2024-10-22-2257 47](https://github.com/user-attachments/assets/60806198-f596-4e4a-a943-172c331eb462)

They specifically are pushing people to the $1.99 a month tier and require a monthly email confirmation every 30 days for the free tier. Frankly I'd recommend just going and buying a cheap domain on porkbun for 10 years, and that'll come out to being less than $1 a month for many TLDs (`.dev`, `.xyz`, `.com`)

**Comment by Mixa84** [comment]
You are probably right, as when I have their updater installed on the machine I still needed to confirm the hostname.

But on the other side, if sometime there is time to do this maybe it would be useful for some DDNS providers to just periodically repeat setting the IP.

Thnx anyway!
